717aa9cea7 broke use of dtc 1.4.6

```
lilith[773]$ git one | head -1
* 717aa9cea7 (HEAD) [Fri Jan 18 14:10:04 2019 -0600] cmake: dts: Tweak warning flags for unit_address check
lilith[774]$ dtc --version
Version: DTC 1.4.6
lilith[775]$ pwd
/mnt/devel/external/zephyr/samples/sensor/sht3xd/ble400
lilith[776]$ rm -rf * && cmake -GNinja -DBOARD=nrf51_ble400 ..
Zephyr version: 1.13.99
-- Found PythonInterp: /usr/bin/python3 (found suitable version "3.6.7", minimum required is "3.4") 
-- Selected BOARD nrf51_ble400
Parsing Kconfig tree in /mnt/devel/external/zephyr/Kconfig
Loading /mnt/devel/external/zephyr/boards/arm/nrf51_ble400/nrf51_ble400_defconfig as base
Merging /mnt/devel/external/zephyr/samples/sensor/sht3xd/prj.conf
Configuration written to '/mnt/devel/external/zephyr/samples/sensor/sht3xd/ble400/zephyr/.config'
-- Loading /mnt/devel/external/zephyr/boards/arm/nrf51_ble400/nrf51_ble400.dts as base
-- Overlaying /mnt/devel/external/zephyr/dts/common/common.dts
-- Overlaying /mnt/devel/external/zephyr/samples/sensor/sht3xd/nrf51_ble400.overlay
FATAL ERROR: Unrecognized check name "unique_unit_address"
CMake Error at /mnt/devel/external/zephyr/cmake/dts.cmake:103 (message):
  command failed with return code: 1
Call Stack (most recent call first):
  /mnt/devel/external/zephyr/cmake/app/boilerplate.cmake:406 (include)
  CMakeLists.txt:8 (include)


-- Configuring incomplete, errors occurred!
```

--- a/cmake/dts.cmake
+++ b/cmake/dts.cmake
@@ -86,13 +86,18 @@ if(CONFIG_HAS_DTS)
   if (check)
     set(DTC_WARN_UNIT_ADDR_IF_ENABLED "-Wunique_unit_address_if_enabled")
   endif()
+  set(DTC_NO_WARN_UNIT_ADDR "")
+  check_dtc_flag("-Wno-unique_unit_address" check)
+  if (check)
+    set(DTC_NO_WARN_UNIT_ADDR "-Wno-unique_unit_address")
+  endif()
   execute_process(
     COMMAND ${DTC}
     -O dts
     -o ${BOARD}.dts_compiled
     -b 0
     -E unit_address_vs_reg
-    -W no-unique_unit_address
+    ${DTC_NO_WARN_UNIT_ADDR}
     ${DTC_WARN_UNIT_ADDR_IF_ENABLED}
     ${EXTRA_DTC_FLAGS} # User settable
     ${BOARD}.dts.pre.tmp

